### PR TITLE
[openapi] add optional fields required to evaluate NET0470/0260

### PR DESCRIPTION
**Pull request by Shastick**
Fulfills the base requirement for https://github.com/interuss/monitoring/issues/832. 

Adds to the observation API:

 * timestamp accuracy 
 * geodetic vertical accuracy
 * horizontal accuracy
 * speed accuracy
 * vertical speed
 * UA Type 
 * UA Classification Type 

Adds to the injection API:

 * UA Type

The PR also moves a few things to the common file for re-use.

Part of the wider effort on https://github.com/interuss/monitoring/issues/754

### CODE REVIEW

File: rid/v1/observation.yaml [hunk 1]
```diff
@@ -122,12 +126,33 @@ components:
...
+        speed_accuracy:
+          anyOf:
+            - $ref: 'commons.yaml#/components/schemas/SpeedAccuracy'
+          description: Accuracy of horizontal ground speed.
+        vertical_speed:
+          format: float
+          description: Speed up (vertically) WGS84-HAE, m/s.
+          type: number
+          example: 0.2
         # TODO for next version: Move most_recent_position here.
```
Comment (Shastick) [inline rid/v1/observation.yaml:156]: @barroco I ran into this TODO: I don't think it is a priority for the current NetRID effort, but if it has any importance, doing the move could be done along this change in a separate PR.

File: rid/v1/observation.yaml [hunk 2]
```diff
@@ -122,12 +126,33 @@ components:
         timestamp:
           description: Time at which the message containing the current state information was generated by the display provider.
           type: string
+        timestamp_accuracy:
```
Comment (mickmis) [inline rid/v1/observation.yaml:129]: This is also defined in `injection.yaml`: to factor away in `commons.yaml`. Beware however to the default, it looks like in `injection.yaml` the field is taken from v19, while you have put here the field from v22a, there is a difference in the default value.

File: rid/v1/observation.yaml [hunk 3]
```diff
@@ -122,12 +126,33 @@ components:
...
           $ref: 'commons.yaml#/components/schemas/RIDOperationalStatus'
         track:
           $ref: 'commons.yaml#/components/schemas/RIDTrack'
         speed:
           $ref: 'commons.yaml#/components/schemas/RIDSpeed'
+        speed_accuracy:
+          anyOf:
+            - $ref: 'commons.yaml#/components/schemas/SpeedAccuracy'
+          description: Accuracy of horizontal ground speed.
+        vertical_speed:
```
Comment (mickmis) [inline rid/v1/observation.yaml:151]: This is also defined in `injection.yaml`: to factor away in commons.yaml.

File: rid/v1/observation.yaml [hunk 4]
```diff
@@ -187,6 +218,18 @@ components:
           $ref: 'commons.yaml#/components/schemas/Longitude'
         alt:
           $ref: 'commons.yaml#/components/schemas/Altitude'
+        accuracy_h:
+          anyOf:
+            - $ref: 'commons.yaml#/components/schemas/HorizontalAccuracy'
+          description: Horizontal error that is likely to be present in this reported
```
Comment (mickmis) [inline rid/v1/observation.yaml:224]: Update description: there is no extrapolated field.

File: rid/v1/observation.yaml [hunk 5]
```diff
@@ -187,6 +218,18 @@ components:
...
+            position.  Required when `extrapolated` field is true and always in the
+            entry for the current state.
+        accuracy_v:
+          anyOf:
+            - $ref: 'commons.yaml#/components/schemas/VerticalAccuracy'
+          description: Vertical error that is likely to be present in this reported
```
Comment (mickmis) [inline rid/v1/observation.yaml:230]: Update description: there is no extrapolated field.

**Review by mickmis — COMMENTED** [review]
Here is a pass on the PR. In addition:
- I think it is worth splitting that PR in modification to injection interface and observation interface. This may help implementers identify the changes in interfaces for each.
- The version of the interface should be incremented of a minor version.

**Comment by Shastick** [comment]
>     * I think it is worth splitting that PR in modification to injection interface and observation interface. This may help implementers identify the changes in interfaces for each.

Agreed. I did a pass on your comments: I suggest you confirm they are good for you, and I then open two separate PRs for updating injections and observation file respectively?

**Comment by mickmis** [comment]
I take that this PR may be closed?

**Pull request closed without merging**